packaging/os/apt.py

```diff
@@ -595,20 +606,13 @@ def main():
             if latest and '=' in package:
                 module.fail_json(msg='version number inconsistent with state=latest: %s' % package)
 
-        if p['state'] == 'latest':
+        if p['state'] in ('latest', 'present', 'build-dep'):
+            build_dep = p.get('build-dep', False)
```

Comment:
Need to set build_dep depending on state.  Also need to set upgrade depending on whether we use latest or present.  Probable some code like this:

``` python
build_dep = False
upgrade = False
if p['state'] == 'build-dep':
    build_dep = True
elif p['state'] == 'latest':
    upgrade = True
```

Then make use of build_dep and upgrade in the call to install().